tsdate/demography.py

```diff
@@ -144,6 +145,100 @@ def to_coalescent_timescale(self, time_ago):
         )
         return coalescent_time_ago
 
+    # @staticmethod
+    # def _Gamma(z, a=0, b=np.inf):
+    #    """
+    #    log |Re| of generalized incomplete gamma function with bounds `a,b`
+    #    and argument `z`
+    #    """
+    #    val = mpmath.log(mpmath.gammainc(z, a=a, b=b))
+    #    return float(val)
+
+    # def to_gamma(self, shape=1, rate=1):
+    #    """
+    #    Given a gamma distribution on a coalescent timescale with parameters
+    #    `shape` and `rate`, return a gamma approximation to the distribution
+    #    under a change of measure to a generational timescale.
+
+    #    :param float shape: Shape parameter of gamma
+    #    :param float rate: Rate parameter of gamma (inverse of scale)
+    #    :return: Shape and rate parameters after change of measure
+    #    """
+    #    assert shape > 0, "Gamma shape parameter must be positive"
+    #    assert rate > 0, "Gamma rate parameter must be positive"
+    #    C = shape * np.log(rate) - self._Gamma(shape)
+    #    mn = 0.0
+    #    va = 0.0
+    #    for a, b, n, t in zip(
+    #        self.coalescent_breaks,
+    #        np.append(self.coalescent_breaks[1:], [np.inf]),
+    #        self.population_size,
+    #        self.time_breaks,
+    #    ):
+    #        gamma = [
+    #            np.exp(
+    #                self._Gamma(shape + i, rate * a, rate * b)
+    #                - (shape + i) * np.log(rate)
+    #                + C
+    #            )
+    #            for i in range(3)
+    #        ]
+    #        mn += (t - n * a) * gamma[0] + n * gamma[1]
+    #        va += (
+    #            gamma[0] * (t - n * a) ** 2
+    #            + 2 * gamma[1] * n * (t - n * a)
+    #            + gamma[2] * n**2
+    #        )
+    #    va -= mn**2
+    #    new_shape = mn**2 / va
+    #    new_rate = mn / va
+    #    return new_shape, new_rate
+
+    def to_gamma(self, shape=1, rate=1):
```

Comment:
I don't think we use this in this PR do we? Should we put this into a follow up?

Reply:
It's used here https://github.com/tskit-dev/tsdate/blob/1fbfcba4965c0383ad6ea78742c5effc2576aade/tsdate/prior.py#L1134-L1136

and also has some tests written. It's necessary in fact even when there's no variation in population size, to get the scaling right

Reply:
Oh, sorry, I missed that.